### PR TITLE
Dump + rebuild files/a/0/2/0 (box icons)

### PR DESCRIPTION
**Pull request by lhearachel**

### CODE REVIEW

**Review by red031000 — CHANGES_REQUESTED** [review]
pls dump every single image with -object

File: files/poketool/icongra/poke_icon/poke_icon.mk [hunk 1]
```diff
@@ -0,0 +1,36 @@
+POKE_ICON_DIR := files/poketool/icongra/poke_icon
+POKE_ICON_NARC := $(POKE_ICON_DIR)/poke_icon.narc
+POKE_ICON_BUILD_DIR := $(POKE_ICON_DIR)/build-poke_icon
+
+POKE_ICON_GFX_FLAGS_ICON := -clobbersize -version101
```
Comment (red031000) [inline files/poketool/icongra/poke_icon/poke_icon.mk:5]: these are already defined in graphics_rules.mk

File: files/poketool/icongra/poke_icon/poke_icon.mk [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+POKE_ICON_PAL_OBJS := $(patsubst $(POKE_ICON_DIR)/%.pal,$(POKE_ICON_BUILD_DIR)/%.NCLR,$(POKE_ICON_PAL_FILES))
+POKE_ICON_ANIM_FILES := $(POKE_ICON_DIR)/0_00000001.json $(POKE_ICON_DIR)/0_00000003.json $(POKE_ICON_DIR)/0_00000005.json
+POKE_ICON_ANIM_OBJS := $(patsubst $(POKE_ICON_DIR)/%.json,$(POKE_ICON_BUILD_DIR)/%.NANR,$(POKE_ICON_ANIM_FILES))
+POKE_ICON_CELL_FILES := $(POKE_ICON_DIR)/0_00000002.json $(POKE_ICON_DIR)/0_00000004.json $(POKE_ICON_DIR)/0_00000006.json
+POKE_ICON_CELL_OBJS := $(patsubst $(POKE_ICON_DIR)/%.json,$(POKE_ICON_BUILD_DIR)/%.NCER,$(POKE_ICON_CELL_FILES))
+POKE_ICON_ICON_FILES := $(wildcard $(POKE_ICON_DIR)/*.png)
+POKE_ICON_ICON_OBJS := $(patsubst $(POKE_ICON_DIR)/%.png,$(POKE_ICON_BUILD_DIR)/%.NCGR,$(POKE_ICON_ICON_FILES))
+
+$(POKE_ICON_BUILD_DIR)/%.NCLR: $(POKE_ICON_DIR)/%.pal
+	@mkdir -p $(POKE_ICON_BUILD_DIR)
```
Comment (red031000) [inline files/poketool/icongra/poke_icon/poke_icon.mk:18]: why not build in place?
alternatively, shove these in /build

File: tools/nitrogfx/main.c [hunk 3]
```diff
@@ -288,7 +288,7 @@ void HandleNtrToPngCommand(char *inputPath, char *outputPath, int argc, char **a
             if (!ParseNumber(argv[i], NULL, 10, &options.palIndex))
                 FATAL_ERROR("Failed to parse palette index.\n");
 
-            if (options.width < 1)
+            if (options.palIndex < 1)
```
Comment (red031000) [inline tools/nitrogfx/main.c:291]: would be appreciated if this was added to red031000/nitrogfx as well

**Review by red031000 — CHANGES_REQUESTED** [review]

File: tools/nitrogfx/main.c (hunk 3 above)
Comment (lhearachel) [reply tools/nitrogfx/main.c:291]: I made a PR for this back in September: https://github.com/red031000/nitrogfx/pull/6

File: files/poketool/icongra/poke_icon/poke_icon.mk (hunk 1 above)
Comment (lhearachel) [reply files/poketool/icongra/poke_icon/poke_icon.mk:5]: Trying to use them legitimately breaks the entire build. I'm not sure exactly what I did wrong to make them not be visible, but they are functionally unusable.

File: files/poketool/icongra/poke_icon/poke_icon.mk (hunk 2 above)
Comment (lhearachel) [reply files/poketool/icongra/poke_icon/poke_icon.mk:18]: `build/` doesn't have any requisite folder structure from what I can see, and building in-place is bad practice. This feels out-of-scope.

Comment (red031000) [reply files/poketool/icongra/poke_icon/poke_icon.mk:18]: disagree, building in place is the default, and should be adhered to, changing that should be done separately

Comment (lhearachel) [reply files/poketool/icongra/poke_icon/poke_icon.mk:18]: I don't know that the project can call it a default when some existing targets don't do it (this one is modeled after the target for `pokegra`, which also builds to a target directory). That being said, I can understand wanting to target an initiative; my question then is: what *is* the initiative? Do you want the project to always build in-place, or build to a target folder?

Comment (red031000) [reply files/poketool/icongra/poke_icon/poke_icon.mk:18]: ideally in place, everything is already set up for that

File: files/poketool/icongra/poke_icon/.gitignore [hunk 4]
```diff
@@ -0,0 +1,5 @@
+*.NCLR
+*.NCGR
+*.NCER
+*.NANR
+build-poke_icon/
```
Comment (PikalaxALT) [inline files/poketool/icongra/poke_icon/.gitignore:5]: Add newline at end of file

File: files/poketool/icongra/poke_icon/0_00000000.pal [hunk 5]
```diff
@@ -0,0 +1,259 @@
+JASC-PAL
```
Comment (PikalaxALT) [inline files/poketool/icongra/poke_icon/0_00000000.pal:1]: Rename to `poke_icon_00000000.pal` (same for remaining files)

**Review by PikalaxALT — CHANGES_REQUESTED** [review]
Other changes: Rename enum `NARC_a_0_2_0` in `include/filesystem_files_def.h`

File: files/poketool/icongra/poke_icon/poke_icon.mk (hunk 1 above)
Comment (red031000) [reply files/poketool/icongra/poke_icon/poke_icon.mk:5]: see pika's PR for an example on how to do this ( #296 )

Comment (lhearachel) [reply files/poketool/icongra/poke_icon/poke_icon.mk:4]: I tried adding these lines, but that just broke my targets, and the NARC would no longer build:

```
CLOBBER_SIZE_VERSION101_NCGR_FILES += $(POKE_ICON_ICON_OBJS)
4BPP_NCLR_FILES += $(POKE_ICON_PAL_OBJS)
```

As best as I can tell, these targets are unused save for a single instance in the Unown puzzle app build, so I would really rather this not be what holds up this PR.

**Review by red031000 — APPROVED** [review]
pls merge my nitrogfx PR before this one

**Review by PikalaxALT — APPROVED** [review]

**Pull request merged**